lib/common/output_xml.c

```diff
@@ -158,6 +158,26 @@ xml_subprocess_output(pcmk__output_t *out, int exit_status,
     free(rc_as_str);
 }
 
+static void
+xml_version(pcmk__output_t *out, bool extended) {
+    xmlNodePtr node;
+    xml_private_t *priv = out->priv;
+    CRM_ASSERT(priv != NULL);
+
+    node = xmlNewChild(g_queue_peek_tail(priv->parent_q), NULL, (pcmkXmlStr) "version", NULL);
+
+    if (extended) {
+        xmlSetProp(node, (pcmkXmlStr) "program", (pcmkXmlStr) "Pacemaker");
+        xmlSetProp(node, (pcmkXmlStr) "version", (pcmkXmlStr) PACEMAKER_VERSION);
+        xmlSetProp(node, (pcmkXmlStr) "build", (pcmkXmlStr) BUILD_VERSION);
+        xmlSetProp(node, (pcmkXmlStr) "features", (pcmkXmlStr) CRM_FEATURES);
+    } else {
+        xmlSetProp(node, (pcmkXmlStr) "program", (pcmkXmlStr) "Pacemaker");
+        xmlSetProp(node, (pcmkXmlStr) "version", (pcmkXmlStr) PACEMAKER_VERSION);
+        xmlSetProp(node, (pcmkXmlStr) "author", (pcmkXmlStr) "Andrew Beekhof");
+    }
```

Comment:
I see you were a step ahead of me :)

The XML formatter should ignore "extended" and just put everything in there. In general, XML format should ignore anything like "quiet", "verbose", "extended", etc. since it's meant to be machine-readable. HTML on the other hand could theoretically make use of such things.

Reply:
Yeah, might as well do it while I'm here and thinking about it.  If I put it off until later, there's a chance I'll forget.